### DISCUSSION TITLE
issue with install Traffic Light Enhancer

### DISCUSSION THREAD

**Discussion by elricks** [slyh/Cities2-TrafficLightsEnhancement#14] [Q&A]
I am a new user of this Mod. Old player of CS1. and have been suing CS2 from day 1.  I have an issue with Traffic Light Enhancer, but before I put all the data here I want to be sure I am in the right place.  Please confirm, or please give me an alternate way to show you what my issue is.

**Comment by slyh** [comment]
You are welcome to open an issue if you believe the mod is not behaving correctly or as intended.

If you need technical assistance, you could ask for help here.

**Reply by elricks** [reply]
Thank you.  I am uncertain if it is me or the app.  I will give you as much info as I can.

I followed Biffa's instructions.

Window 10.

I am using BepinExPaxk 5.4.2100
I have other mods installed previously
a, Unlock all tiles (confirmed as working)
First Person Camera (installed, not used as yet)
Legacy Flavour (installed, but I have yet to find what is required to use it)

In Thunderstore 1.3.4.0 I see all the mods listed above and all show as active. This includes Traffic Lights Enhanced, which shows 5.0.0 in the right-most field..

I get no errors, but when I choose the 'traffic lights. icon I do not get a pop up with options.  The intersection does highlight in blue.

My existing game (Busy Town, all tiles owned (per unlock all tiles)),
I first tried Traffice Lights Enhanced on  existing game with existing road.
I then tried creating new intersections within the existing game.  3 and 4 lane roads.
I then tried a brand new game, but this failed as I would need to play on somewhat to get access to the option.

I am happy to send any config files, just ask.

I am raising this in the spirit of assisting you and others to identify potential errors. I am NOT clamouring to get a fix for myself. Please give this a priority that suits you and your timetable.

SHIRLEY

**Reply by slyh** [reply]
It's very likely that the mod failed to load. If the panel doesn't appear, it would suggest that the mod isn't loading properly.

Normally, you should see messages similar to the following in your BepInEx console:
```
[Info   :   BepInEx] Loading [C2VM.CommonLibraries.LaneSystem 0.0.1]
[Info   :C2VM.CommonLibraries.LaneSystem] Plugin C2VM.CommonLibraries.LaneSystem 0.0.1 is loaded!
[Info   :   BepInEx] Loading [C2VM.TrafficLightsEnhancement 0.2.0]
[Info   :C2VM.TrafficLightsEnhancement] Plugin C2VM.TrafficLightsEnhancement 0.2.0 is loaded!
```

Could you tell me the exact link you used to download the mod? Also, could you provide the log file located at BepInEx/LogOutput.log?

**Reply by elricks** [reply]
One extra piece of info.  Right mouse click removes traffic lights, left places them, which I think is the default without a mod installed?

**Reply by elricks** [reply]
I am sure you are correct, it did not load.  Even though I am OK at using file manager, and in installing and uninstalling all sorts of PC apps, I am unable to find the BepInEx folder etc.  Please don't hassle, I am happy to wait until you mod makes it to the Thunderstore.  I really want to use this, so rest assured I will be keeping a really good eye on the available mods.  Consider this as CLOSED.

**Reply by slyh** [reply]
Sorry for keeping you waiting, I think the mod will be ready to be released on Thunderstore soon.

**Comment by elricks** [comment]
Sorry I added my last without seeing your reply....Thanks for taking the time to communicate

[LogOutput.log](https://github.com/slyh/Cities2-Various-Mods/files/13463453/LogOutput.log)

I clicked on the link to version 5 on this page - https://github.com/slyh/Cities2-Various-Mods

Download the mod

For BepInEx 5 users, download [C2VM.TrafficLightsEnhancement-BepInEx5.zip](https://github.com/slyh/Cities2-Various-Mods/releases/latest/download/C2VM.TrafficLightsEnhancement-BepInEx5.zip)

I followed BIFFA'S instructions (watched his youtube, stopping where appropriate, during my install process).  I notice that you say to unzip, he says not to.  I did NOT.  I think I should uninstall, then reinstall using your instructions - BUT I WILL AWAIT your next comments before proceeding.

BUT the install process gave no error messages.

**Reply by MadSynth** [reply]
Any resolution for this yet? I tried installing it on my Steam Deck, but the mod didn't seem to load in Game or Desktop mode. I realize it's probably not a supported platform for the mod, but it's the only machine I have that will run the game stable. Hope I can get it working!

**Reply by slyh** [reply]
According to the log file, BepInEx did not attempt to load the mod. Perhaps the mod is disabled in the mod manager?

Regarding running mods on Steam Deck, I assume you are using SteamOS. You will need to set a DLL override for Wine.

You could refer to the following guides:
https://docs.bepinex.dev/articles/advanced/proton_wine.html
https://steamcommunity.com/sharedfiles/filedetails/?id=2841673787

**Reply by MadSynth** [reply]
I was able to get it working on both my PC and my Steam Deck. I rechecked the game files on Steam, and found there was a file missing on my PC, and let it update, then reinstalled the mod from scratch, and it works fine now. I did the same thing with the Steam Deck, but had to add a line of text to LAUNCH OPTIONS under the Cities: Skylines II settings.

`WINEDLLOVERRIDES='winhttp.dll=n,b' %command%`

 Thanks for the tips!
